sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -165,10 +165,18 @@ case class StateStoreMetrics(
 
 object StateStoreMetrics {
   def combine(allMetrics: Seq[StateStoreMetrics]): StateStoreMetrics = {
+    val customMetricNames = allMetrics.flatMap(_.customMetrics.keySet).map(_.name).toSet
```

Comment:
nit: why not use  `StateStoreCustomMetric` to filter the same metric directly?

Reply:
Something like this?

```scala
val distinctCustomMetrics = allMetrics.flatMap(_.customMetrics.keys).distinct
val customMetrics = allMetrics.flatMap(_.customMetrics)
val combinedCustomMetrics = distinctCustomMetrics.map { customMetric =>
  val sameMetrics = customMetrics.filter(_ == customMetric)
  val sumOfMetrics = sameMetrics.map(_._2).sum
  customMetric -> sumOfMetrics
}.toMap
```

Reply:
Hmm, although the some custom metrics like `StateStoreCustomSumMetric` are case class, the base trait `StateStoreCustomMetric` can also be implemented in class too. So currently comparing ` StateStoreCustomMetric` directly looks okay, I'm not sure if it is possible to fail if someone implements a custom metric with class, not case class.

Reply:
Maybe you can override `equal()` for the base trait if you have such concern.